### PR TITLE
feat: Sync API with Windows SDK 22000

### LINKED ISSUE #8374: Update API definitions to latest UWP
## What would you like to be added:

We are currently mostly on 17763 API surface-wise, would be nice to add the latest APIs, so that porting UWP apps is easier.

## Why is this needed:

UWP parity.

## For which Platform:

All

### LINKED ISSUE #8376: Sync generator does not properly generate for Composition and Dispatching in UWP mode
### Current behavior

For `Composition` and `Dispatching` the Sync generator is not properly generating the API, as it is not using the compilation in those projects.

### Expected behavior

Should work

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #8374, closes #8376

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

API matching 17763

## What is the new behavior?

API matching 22000

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI.Composition/Generated/3.0.0.0/Windows.UI.Composition.Interactions/VisualInteractionSource.cs [hunk 1]
```diff
@@ -232,13 +232,7 @@ public  float ScaleVelocity
 		// Forced skipping of method Windows.UI.Composition.Interactions.VisualInteractionSource.ScaleSourceMode.get
 		// Forced skipping of method Windows.UI.Composition.Interactions.VisualInteractionSource.ScaleSourceMode.set
 		// Forced skipping of method Windows.UI.Composition.Interactions.VisualInteractionSource.Source.get
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
```
Comment (MartinZikmund) [inline src/Uno.UI.Composition/Generated/3.0.0.0/Windows.UI.Composition.Interactions/VisualInteractionSource.cs:235]: This is weird 🤔🤔🤔

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
+		public static global::Windows.UI.Core.CoreIndependentInputSourceController CreateForVisual( global::Windows.UI.Composition.Visual visual)
+		{
+			throw new global::System.NotImplementedException("The member CoreIndependentInputSourceController CoreIndependentInputSourceController.CreateForVisual(Visual visual) is not implemented in Uno.");
+		}
+		#endif
+		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
+		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
+		public static global::Windows.UI.Core.CoreIndependentInputSourceController CreateForIVisualElement( global::Windows.UI.Composition.IVisualElement visualElement)
```
Comment (MartinZikmund) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs:83]: @jeromelaban this is going to be interesting - this class is using `Windows.UI.Composition` but it is in "`Uno.UWP`" 🤔

Comment (MartinZikmund) [reply src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs]: For now I just commented out these two methods, as they are unlikely to be used 😁

File: src/Uno.UI.Composition/Generated/3.0.0.0/Windows.UI.Composition.Interactions/VisualInteractionSource.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI.Composition/Generated/3.0.0.0/Windows.UI.Composition.Interactions/VisualInteractionSource.cs:235]: Fixed

Comment (Youssef1313) [reply src/Uno.UI.Composition/Generated/3.0.0.0/Windows.UI.Composition.Interactions/VisualInteractionSource.cs:235]: It's skipped in 

https://github.com/unoplatform/uno/blob/87084f8d6a5c258fd6f95ea5139449dbf36e0d10/src/Uno.UWPSyncGenerator/Generator.cs#L1175-L1183

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs (hunk 2 above)
Comment (Youssef1313) [reply src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs]: @MartinZikmund Could you please avoid manual changes to generated files? They're very likely to get reverted unintentionally when someone comes later and run the generator.

You can skip methods here:

https://github.com/unoplatform/uno/blob/87084f8d6a5c258fd6f95ea5139449dbf36e0d10/src/Uno.UWPSyncGenerator/Generator.cs#L1123

**Review by Youssef1313 — COMMENTED** [review]
This is very nice!! 🎉 

Hope this unblocks #7942 where I'm having issues related to Composition.

File: src/Uno.UWPSyncGenerator/Generator.cs [hunk 3]
```diff
@@ -122,6 +124,26 @@ where Path.GetFileNameWithoutExtension(externalRedfs.Display).StartsWith("Window
...
+			if (baseName == "Uno.UI.Dispatching")
+			{
+				excludeNamespaces.Clear();
+				includeNamespaces.Add("Windows.UI.Dispatching");
+			}
+			if (baseName == "Uno.UI.Composition")
+			{
+				excludeNamespaces.Clear();
+				includeNamespaces.Add("Windows.UI.Composition");
+			}
```
Comment (Youssef1313) [inline src/Uno.UWPSyncGenerator/Generator.cs:144]: I feel this is more readable. But it's a minor change so feel free if you don't want to take it :)

```suggestion
			if (baseName == "Uno.UI.Dispatching")
			{
				includeNamespaces.Add("Windows.UI.Dispatching");
			}
			else if (baseName == "Uno.UI.Composition")
			{
				includeNamespaces.Add("Windows.UI.Composition");
			}
			else
			{
				excludeNamespaces.Add("Windows.UI.Dispatching");
				excludeNamespaces.Add("Windows.UI.Composition");
			}
```

**Review by Youssef1313 — COMMENTED** [review]
Can you verify the generator is still tracking this directory?

https://github.com/unoplatform/uno/tree/873fb2ed986c124e972b9bf7b5361ca87a757881/src/Uno.UI.Composition/Generated/3.0.0.0

e.g, make a bad change and run it to see if updates it?

I'm asking because I see:

https://github.com/unoplatform/uno/blob/87084f8d6a5c258fd6f95ea5139449dbf36e0d10/src/Uno.UWPSyncGenerator/Program.cs#L42-L45

those lines are in `HAS_UNO_WINUI`. I assume this directory should always be written?

File: src/Uno.UWPSyncGenerator/Generator.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UWPSyncGenerator/Generator.cs]: Yes, that's probably nicer, amended to last commit 👍

**Comment by MartinZikmund** [comment]
> Can you verify the generator is still tracking this directory?
> 
> https://github.com/unoplatform/uno/tree/873fb2ed986c124e972b9bf7b5361ca87a757881/src/Uno.UI.Composition/Generated/3.0.0.0
> 
> e.g, make a bad change and run it to see if updates it?
> 
> I'm asking because I see:
> 
> https://github.com/unoplatform/uno/blob/87084f8d6a5c258fd6f95ea5139449dbf36e0d10/src/Uno.UWPSyncGenerator/Program.cs#L42-L45
> 
> those lines are in `HAS_UNO_WINUI`. I assume this directory should always be written?

Yes it does generate correctly now (at least in the UWP mode which I tested). For WinUI mode there should be no changes (as the include/exclude is UWP-mode specific).

**Comment by Youssef1313** [comment]
@MartinZikmund I'm very surprised it works. You check for `baseName == `"Uno.UI.Composition", but the only callers that pass this `baseName` are under WinUI condition.

https://github.com/unoplatform/uno/blob/87084f8d6a5c258fd6f95ea5139449dbf36e0d10/src/Uno.UWPSyncGenerator/Program.cs#L36-L43

Why does it work? 😄

**Comment by MartinZikmund** [comment]
@Youssef1313 I actually added them in my branch for the UWP mode (at the end of the method) - 
https://github.com/unoplatform/uno/blob/94afbca559f9887430c7ba2405ad74f3e82a80b8/src/Uno.UWPSyncGenerator/Program.cs#L58-L59

But maybe same change should be done for the docs generator as well (currently only Sync)

**Comment by Youssef1313** [comment]
@MartinZikmund Ah sorry didn't see this change. I think I was looking at the diff for a single commit thinking it's the whole PR 😄

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs (hunk 2 above)
Comment (Youssef1313) [reply src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Core/CoreIndependentInputSourceController.cs]: @MartinZikmund This comment is marked as resolved but I see the last commit still have this part commented.

**Comment by jeromelaban** [comment]
We can rerun once #8343 is merged.

**Comment by MartinZikmund** [comment]
@jeromelaban Done and ready 👍

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**